src/main/java/seedu/address/model/person/SalaryRange.java

```diff
@@ -0,0 +1,175 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+
+/**
+ * Represents a Person's salary in the address book.
+ */
+public class SalaryRange {
+    public static final String MESSAGE_CONSTRAINTS = "SALARY_RANGE should only contain numbers, with range [0, "
+            + "4294967295] "
+            + "or two pure digital numbers with ‘-’ in between or '<=' or '>=' followed by a digital number.\n"
+            + "Numbers can vary from large to small or from small to large.\n"
+            + "Both digital numbers should be within the range [0, 2147483647]\n"
+            + "Examples:\n"
+            + "2000-4000\n"
+            + ">=5000\n"
+            + "<=10000\n";
+    public static final String VALIDATION_REGEX = "^(\\d+-\\d+)|((<=|>=)\\d+)|(\\d+)$";
+    public static final int UPPERBOUND = 2147483647;
+    public static final int LOWERBOUND = 0;
+
+    private int maxSalary;
+    private int minSalary;
+    private boolean isRange;
+    private boolean isMax;
+    private boolean isMin;
+
+    /**
+     * Constructs a {@code Salary}.
+     *
+     * @param salaryRange A valid salary range.
+     */
+    public SalaryRange(String salaryRange) {
+        requireNonNull(salaryRange);
+        checkArgument(isValidSalaryRange(salaryRange), MESSAGE_CONSTRAINTS);
+        parseSalaryRange(salaryRange);
+    }
+
+    /**
+     * Constructs a {@code Salary}.
+     *
+     * @param salaryRange A valid salary in string format.
+     */
+    public void parseSalaryRange(String salaryRange) {
```

Comment:
There is some code duplication in the way different cases are handled. Might want to consider refactoring to reduce redundancy.